### DISCUSSION TITLE
Questions about new Vercel directory setup / api pages / pages / router

### DISCUSSION THREAD

**Discussion by Elindo586** [vercel/next.js#67854] [Help] [answered]
### Summary

So, I don't get to code new projects all that much, so I have questions about the new directory.

In image below. the file "emails" is a new project, and I noticed the folder "app" instead of "pages" from previous versions, and the file "page.js" instead of  "index.js"

Also, I believe Next.js used to have the folder `pages/api`  and I don't see that anymore.   Where could I put the api folder?  and can I still write code same as this? 

```
export default function handler(req, res) {
  res.status(200).json({ name: 'John Doe' })
}
```

Also... the way Next.js used to work is that the what I call the "routes" would be created depending on the file structure of the directory.... what is the purpose of the `router` ??   In the new installation I was asked if I wanted the App Router and I said yes.

Here is an image of a new "emails" directory vs. an older "nextprojectb" directory which includes pages/api folder

![image](https://github.com/user-attachments/assets/c6e8984e-4207-492a-9c8c-f015ff39eed1)

**Comment by icyJoseph** [comment, marked as answer]
> ### Summary
> So, I don't get to code new projects all that much, so I have questions about the new directory.
> 
> In image below. the file "emails" is a new project, and I noticed the folder "app" instead of "pages" from previous versions, and the file "page.js" instead of "index.js"
> 

The `app` folder is for the App Router tech stack, which uses React Server Components, and it is totally detached, in terms of the JavaScript context, from the Pages Router, which used to into `pages` folder.

In the app folder, the file structure is still what decides the routes available for people to consume, but rather than just picking up on every javascript file under app (roughly speaking), as pages folder did, app router picks up on special files:

- page.js
- loading.js
- layout.js
- route.js

Those are, I'd say the, absolute essential, with their TypeScript counterparts. There are a few more additionally.

You need to learn and read about their differences, and when each is useful.

Follow this section of the docs, https://nextjs.org/docs/app/building-your-application/routing, and the next steps (other items in the docs sub tree)

> Also, I believe Next.js used to have the folder `pages/api` and I don't see that anymore. Where could I put the api folder? and can I still write code same as this?
> 
> ```
> export default function handler(req, res) {
>   res.status(200).json({ name: 'John Doe' })
> }
> ```

You can define now `route.js` files, say `app/my/service/route.js`, would make it so that I could fire a requet to `https://example.com/my/service` 

For example if:

```js
// app/my/service/route.js`

export const GET = () => {
    return new Response('Hello, Next.js!', {
    status: 200,
  })
}
```

These are called, Route Handlers. In Pages Router, these were called API Handlers. These don't need to be inside an special `api` folder anymore.

Unlike in api routes, route handlers in app router, the one I defined above, would allow only GET requests, by default, you'd need to export a `POST` function, to handle POST requests.

Read more about these here, https://nextjs.org/docs/app/building-your-application/routing/route-handlers

> Also... the way Next.js used to work is that the what I call the "routes" would be created depending on the file structure of the directory.... what is the purpose of the `router` ?? In the new installation I was asked if I wanted the App Router and I said yes.
> 

It is the same, but as said above, there are now special file name conventions that you need to learn.

Also, in App Router, there's no more `_app`, nor `_document`, rather, `app/layout.js`, is the closest thing in terms of similarity, and it is the root application, where the html and body of the application are rendered.

**Reply by Elindo586** [reply]
Few new things to learn.

Also.. I got my first error just by copy and pasting a code from an older project to a new one... I will create a new post.
